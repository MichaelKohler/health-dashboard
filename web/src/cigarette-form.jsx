import React from 'react';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

import getDefaultStyle from './styles';
import { postCigarette } from './actions';

const useStyles = makeStyles((theme) => Object.assign({}, getDefaultStyle(theme), {
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
}));

const mapStateToProps = (state) => state;

const mapDispatchToProps = (dispatch) => ({
  postCigarette: () => {
    dispatch(postCigarette());
  },
});

export function CigaretteForm(props) {
  const classes = useStyles();

  return (
      <Container component="main" maxWidth="xs">
          <CssBaseline/>
          <div className={ classes.appBarSpacer }/>
          <div className={ classes.paper }>
              <Typography component="h1" variant="h5">
                  Add Cigarette
              </Typography>
              <form className={ classes.form } noValidate onSubmit={ props.postCigarette }>
                  <FormControlLabel
                      control={
                          <Checkbox
                            id="rolled"
                            checked
                            color="primary"
                            value="checked"
                          />
                      }
                      label="rolled"
                  />
                  <Button
                    type="submit"
                    fullWidth
                    variant="contained"
                    color="primary"
                    className={ classes.submit }
                    disabled={ props.isSubmitting }
                  >
                    Add
                  </Button>
              </form>
          </div>
      </Container>
  );
}

CigaretteForm.propTypes = {
  postCigarette: PropTypes.func,
  isSubmitting: PropTypes.bool,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withRouter(CigaretteForm));
