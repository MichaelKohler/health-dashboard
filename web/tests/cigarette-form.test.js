/* global it, expect */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import renderer from 'react-test-renderer';
import { CigaretteForm } from '../src/cigarette-form.jsx';

it('renders correctly', () => {
  const tree = renderer
    .create(<CigaretteForm/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders button disabled', () => {
  const tree = renderer
    .create(<CigaretteForm isSubmitting={true}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* eslint-enable react/jsx-filename-extension */
