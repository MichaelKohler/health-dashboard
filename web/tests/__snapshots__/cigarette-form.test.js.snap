// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button disabled 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <CssBaseline />
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <div
    className="makeStyles-paper-3"
  >
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Add Cigarette
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-5"
      noValidate={true}
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        control={
          <WithStyles(ForwardRef(Checkbox))
            checked={true}
            color="primary"
            id="rolled"
            value="checked"
          />
        }
        label="rolled"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-6"
        color="primary"
        disabled={true}
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Add
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <CssBaseline />
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <div
    className="makeStyles-paper-3"
  >
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Add Cigarette
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-5"
      noValidate={true}
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        control={
          <WithStyles(ForwardRef(Checkbox))
            checked={true}
            color="primary"
            id="rolled"
            value="checked"
          />
        }
        label="rolled"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-6"
        color="primary"
        disabled={false}
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Add
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;
